### ISSUE TITLE
Error: An update to Mazda app is required to continue

### ISSUE THREAD

**Issue by pr0fetul** [bdr99/pymazda#16]
Hi @bdr99 ,

It seems that Mazda has again updated something related to the MyMazda app, and it stopped working again. Here is the log:
```
Logger: homeassistant.components.mazda
Source: components/mazda/__init__.py:42
Integration: Mazda Connected Services (documentation, issues)
First occurred: 2:06:42 AM (149 occurrences)
Last logged: 9:30:48 AM

Unknown error occurred during Mazda update request: Request failed: An update to Mazda app is required to continue
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 148, in async_update_data
    vehicle["status"] = await with_timeout(
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 42, in with_timeout
    return await task
  File "/usr/local/lib/python3.9/site-packages/pymazda/client.py", line 64, in get_vehicle_status
    vehicle_status_response = await self.controller.get_vehicle_status(vehicle_id)
  File "/usr/local/lib/python3.9/site-packages/pymazda/controller.py", line 31, in get_vehicle_status
    response = await self.connection.api_request("POST", "remoteServices/getVehicleStatus/v4", body_dict=post_body, needs_keys=True, needs_auth=True)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 158, in api_request
    return await self.__api_request_retry(method, uri, query_dict, body_dict, needs_keys, needs_auth, num_retries=0)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 173, in __api_request_retry
    return await self.__send_api_request(method, uri, query_dict, body_dict, needs_keys, needs_auth)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 245, in __send_api_request
    raise MazdaException("Request failed: " + response_json["error"])
pymazda.exceptions.MazdaException: Request failed: An update to Mazda app is required to continue
```

**Comment by bdr99** [comment]
Thanks @pr0fetul. It looks like Mazda made a change to their API to correspond with a new update to the MyMazda mobile app. I think I might just need to increase the app version number in the HTTP headers. I'll look into this.

**Comment by pr0fetul** [comment]
Thank you very much.

**Comment by bdr99** [comment]
I just published version 0.2.1 which should fix this. @pr0fetul Can you please test and let me know?

**Comment by pr0fetul** [comment]
Hi @bdr99,

I downloaded and moved the mazda 2.1 folder to the custom_components folder, restarted HA, but unfortunately I get the same error:
```
Logger: homeassistant.components.mazda
Source: components/mazda/__init__.py:42
Integration: Mazda Connected Services (documentation, issues)
First occurred: 3:10:45 PM (2 occurrences)
Last logged: 3:11:26 PM

Unknown error occurred during Mazda update request: Request failed: An update to Mazda app is required to continue
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 143, in async_update_data
    vehicles = await with_timeout(mazda_client.get_vehicles())
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 42, in with_timeout
    return await task
  File "/usr/local/lib/python3.9/site-packages/pymazda/client.py", line 27, in get_vehicles
    vec_base_infos_response = await self.controller.get_vec_base_infos()
  File "/usr/local/lib/python3.9/site-packages/pymazda/controller.py", line 21, in get_vec_base_infos
    return await self.connection.api_request("POST", "remoteServices/getVecBaseInfos/v4", body_dict={"internaluserid": "__INTERNAL_ID__"}, needs_keys=True, needs_auth=True)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 158, in api_request
    return await self.__api_request_retry(method, uri, query_dict, body_dict, needs_keys, needs_auth, num_retries=0)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 173, in __api_request_retry
    return await self.__send_api_request(method, uri, query_dict, body_dict, needs_keys, needs_auth)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 245, in __send_api_request
    raise MazdaException("Request failed: " + response_json["error"])
pymazda.exceptions.MazdaException: Request failed: An update to Mazda app is required to continue
```

**Comment by bdr99** [comment]
Actually, that's not the correct process to test the new version with HA. I packaged the new version as a custom component here: 
[mazda.zip](https://github.com/bdr99/pymazda/files/7111670/mazda.zip)

If you download and extract this into your custom_components folder, it should load the new version of pymazda. You will see an entry like this in your HA logs to confirm the custom integration is being loaded:

```
2021-09-05 09:23:49 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration mazda which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
```

**Comment by pr0fetul** [comment]
I am sorry - I still have not setup a test environment yet. 
I downloaded the attached folder, put it into custom_components, restarted HA and now it works again.
You were right and it needed a version upgrade.

**Comment by bdr99** [comment]
No worries! Thanks for testing. I'll close this issue since it's fixed. I submitted a PR to Home Assistant to fix the issue there: https://github.com/home-assistant/core/pull/55820

**Comment by pr0fetul** [comment]
Hi Brandon (@bdr99 ) 
It seems that there is another MyMazda update that's needed. 
Can you re-update the code?

```
Logger: homeassistant.components.mazda
Source: components/mazda/__init__.py:42
Integration: Mazda Connected Services (documentation, issues)
First occurred: 9:26:51 AM (51 occurrences)
Last logged: 10:32:11 AM

Unknown error occurred during Mazda update request: Request failed: An update to Mazda app is required to continue
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 143, in async_update_data
    vehicles = await with_timeout(mazda_client.get_vehicles())
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 42, in with_timeout
    return await task
  File "/usr/local/lib/python3.9/site-packages/pymazda/client.py", line 27, in get_vehicles
    vec_base_infos_response = await self.controller.get_vec_base_infos()
  File "/usr/local/lib/python3.9/site-packages/pymazda/controller.py", line 21, in get_vec_base_infos
    return await self.connection.api_request("POST", "remoteServices/getVecBaseInfos/v4", body_dict={"internaluserid": "__INTERNAL_ID__"}, needs_keys=True, needs_auth=True)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 158, in api_request
    return await self.__api_request_retry(method, uri, query_dict, body_dict, needs_keys, needs_auth, num_retries=0)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 173, in __api_request_retry
    return await self.__send_api_request(method, uri, query_dict, body_dict, needs_keys, needs_auth)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 245, in __send_api_request
    raise MazdaException("Request failed: " + response_json["error"])
pymazda.exceptions.MazdaException: Request failed: An update to Mazda app is required to continue
```

Thank you very much

**Comment by pr0fetul** [comment]
Hi Brandon (@bdr99 )
It seems that there is another MyMazda update that's needed.
Can you re-update the code?
```
Logger: homeassistant.components.mazda
Source: components/mazda/__init__.py:43
Integration: Mazda Connected Services (documentation, issues)
First occurred: 3:03:19 AM (108 occurrences)
Last logged: 8:24:48 AM

Unknown error occurred during Mazda update request: Request failed: An update to Mazda app is required to continue
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 148, in async_update_data
    vehicle["status"] = await with_timeout(
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 43, in with_timeout
    return await task
  File "/usr/local/lib/python3.9/site-packages/pymazda/client.py", line 64, in get_vehicle_status
    vehicle_status_response = await self.controller.get_vehicle_status(vehicle_id)
  File "/usr/local/lib/python3.9/site-packages/pymazda/controller.py", line 31, in get_vehicle_status
    response = await self.connection.api_request("POST", "remoteServices/getVehicleStatus/v4", body_dict=post_body, needs_keys=True, needs_auth=True)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 158, in api_request
    return await self.__api_request_retry(method, uri, query_dict, body_dict, needs_keys, needs_auth, num_retries=0)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 173, in __api_request_retry
    return await self.__send_api_request(method, uri, query_dict, body_dict, needs_keys, needs_auth)
  File "/usr/local/lib/python3.9/site-packages/pymazda/connection.py", line 245, in __send_api_request
    raise MazdaException("Request failed: " + response_json["error"])
pymazda.exceptions.MazdaException: Request failed: An update to Mazda app is required to continue
```

**Comment by bdr99** [comment]
@pr0fetul This is now fixed and a PR has been submitted to Home Assistant. See https://github.com/bdr99/pymazda/issues/20#issuecomment-1022773453.

**Comment by adamhyde57** [comment]
Hi Brandon, @bdr99

Not sure if you are still monitoring this but: Mazda app is down again in Home Assistant...

raise MazdaException("Request failed: " + response_json["error"])
pymazda.exceptions.MazdaException: Request failed: An update to Mazda app is required to continue
2023-03-27 21:46:02.204 ERROR (MainThread) [homeassistant.components.mazda] Unknown error occurred during Mazda update request: Request failed: An update to Mazda app is required to continue
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 156, in async_update_data
    vehicle["status"] = await with_timeout(
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 54, in with_timeout
    return await task
  File "/usr/local/lib/python3.10/site-packages/pymazda/client.py", line 65, in get_vehicle_status
    vehicle_status_response = await self.controller.get_vehicle_status(vehicle_id)
  File "/usr/local/lib/python3.10/site-packages/pymazda/controller.py", line 31, in get_vehicle_status
    response = await self.connection.api_request("POST", "remoteServices/getVehicleStatus/v4", body_dict=post_body, needs_keys=True, needs_auth=True)
  File "/usr/local/lib/python3.10/site-packages/pymazda/connection.py", line 162, in api_request
    return await self.__api_request_retry(method, uri, query_dict, body_dict, needs_keys, needs_auth, num_retries=0)
  File "/usr/local/lib/python3.10/site-packages/pymazda/connection.py", line 177, in __api_request_retry
    return await self.__send_api_request(method, uri, query_dict, body_dict, needs_keys, needs_auth)
  File "/usr/local/lib/python3.10/site-packages/pymazda/connection.py", line 251, in __send_api_request

**Comment by bdr99** [comment]
@adamhyde57 Which version of HA are you using?

**Comment by adamhyde57** [comment]
@bdr99 
Home Assistant 2023.3.1
Frontend 20230302.0 - latest

Thanks.

**Comment by bdr99** [comment]
@adamhyde57 Please upgrade to 2023.3.3 or newer to fix this problem.

**Comment by adamhyde57** [comment]
@bdr99 
ok, unfortunately I'm running it in a docker container on my Synology NAS.  3.1 is the latest available, I just tried upgrading.  It was working fine in 3.1 till a few days ago.  Weird.  Guess I'll just have to wait awhile.

**Comment by adamhyde57** [comment]
@bdr99 

Awesome, thanks for the help.  I went back in and it looks like I wasn't scrolling down far enough in the docker registry to find the new version.  I'm on 2023.3.6 and it works!

**Comment by darkonevt777** [comment]
Hi @bdr99,

I run HA version 2023.06.1 and have similiar problems while trying to connect to Mazda service. 
In previous version it worked.

Here is my log:
```
Logger: homeassistant.components.mazda
Source: components/mazda/__init__.py:57
Integration: Mazda Connected Services (documentation, issues)
First occurred: 11:58:20 (4 occurrences)
Last logged: 11:59:19

Unknown error occurred during Mazda update request: 'NoneType' object is not subscriptable
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 159, in async_update_data
    vehicle["status"] = await with_timeout(
                        ^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 57, in with_timeout
    return await task
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymazda/client.py", line 68, in get_vehicle_status
    remote_info = vehicle_status_response.get("remoteInfos")[0]
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
TypeError: 'NoneType' object is not subscriptable
```

**Comment by bdr99** [comment]
@darkonevt777 This is not the same as the original issue. Can you open a new issue to discuss this please?

**Comment by sunishsheth2009** [comment]
I am getting a similar error as well:
```
  File "/usr/local/lib/python3.11/site-packages/pymazda/connection.py", line 252, in __send_api_request
    raise MazdaException("Request failed: " + response_json["error"])
pymazda.exceptions.MazdaException: Request failed: An update to Mazda app is required to continue
2023-07-01 09:33:02.706 ERROR (MainThread) [homeassistant.components.mazda] Unknown error occurred during Mazda update request: Request failed: An update to Mazda app is required to continue
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 154, in async_update_data
    vehicles = await with_timeout(mazda_client.get_vehicles())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/mazda/__init__.py", line 57, in with_timeout
    return await task
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymazda/client.py", line 27, in get_vehicles
    vec_base_infos_response = await self.controller.get_vec_base_infos()
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymazda/controller.py", line 21, in get_vec_base_infos
    return await self.connection.api_request("POST", "remoteServices/getVecBaseInfos/v4", body_dict={"internaluserid": "__INTERNAL_ID__"}, needs_keys=True, needs_auth=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymazda/connection.py", line 163, in api_request
    return await self.__api_request_retry(method, uri, query_dict, body_dict, needs_keys, needs_auth, num_retries=0)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymazda/connection.py", line 178, in __api_request_retry
    return await self.__send_api_request(method, uri, query_dict, body_dict, needs_keys, needs_auth)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymazda/connection.py", line 252, in __send_api_request
    raise MazdaException("Request failed: " + response_json["error"])
pymazda.exceptions.MazdaException: Request failed: An update to Mazda app is required to continue

```

Thanks for your help in advance. :)

**Comment by bdr99** [comment]
@sunishsheth2009 This will be fixed in HA 2023.7.0 which will be released on Wednesday.

**Comment by sunishsheth2009** [comment]
Thank you for the super prompt reply :) Appreciate your help. 
Also side note: Would the application work for Mazda 6 as well?
